File: C# Backend/Student/Student/Model/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace Student.Model
{
    [Table("product", Schema = "public")]
    public class Product
    {
        [Key]
        [Column("id_product")]
        public int Id { get; set; }

        [Column("name")]
        [Required(ErrorMessage = "Name is required")]
        public string Name { get; set; }

        [Column("price")]
        [Required(ErrorMessage = "Price is required")]
        [Range(0, double.MaxValue, ErrorMessage = "Price cannot be negative")]
        public double Price { get; set; }

        [Column("quantity")]
        [Required(ErrorMessage = "Quantity is required")]
        [Range(0, int.MaxValue, ErrorMessage = "Quantity cannot be negative")]
        public int Quantity { get; set; }

        [Column("size")]
        public string Size { get; set; }

        [Column("gender")]
        [RegularExpression("^(man|women)$", ErrorMessage = "Invalid gender. Must be 'man' or 'women'")]
        public string Gender { get; set; }

        public override string ToString()
        {
            return $"Product{{ id={Id}, name='{Name}', price={Price} }}";
        }
    }
}

File: backend/src/main/java/com/commerce/datamodel/Product.java
package com.commerce.datamodel;


import javax.persistence.Access;
import javax.persistence.AccessType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.PositiveOrZero;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(schema = "public", name = "product")
@Getter
@Setter
@Access(AccessType.FIELD)
public class Product
{
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id_product")
  private int id;
  @Column(name = "name")
  @NotBlank(message = "Name is required")
  private String name;
  @Column(name = "price")
  @NotNull(message = "Price is required")
  @PositiveOrZero(message = "Price cannot be negative")
  private Double price;
  
  
  @Column(name = "quantity")
  @NotNull(message = "Quantity is required")
  @Min(value = 0, message = "Quantity cannot be negative")
  private Integer quantity;
  
  @Column(name = "size")
  private String size;
  
  @Column(name = "gender")
  @Pattern(regexp = "^(man|women)$", message = "Invalid gender. Must be 'man' or 'women'")
  private String gender;
  
  
//  add SQL  quantity size and gender
  //quantity XS S M L XL
  //size
  // type man or women
  // positive
//  @Lob
//  @Column(columnDefinition = "MEDIUMBLOB")
//  private String image;
//  @ManyToOne
//  @JoinColumn(name = "id_stock", referencedColumnName = "id_stock")
//  private Stock stock;
  

//  public Product(String name, double price)
//  {
//    this.name = name;
//    this.price = price;
////    this.stock = stock;
//  }
  
  @Override
  public String toString()
  {
    return "Product{" +
           "id=" + id +
           ", name='" + name + '\'' +
           ", price=" + price +
           '}';
  }
  

}
